kafka/src/main/java/io/specmesh/kafka/ProvisionStatus.java

```diff
@@ -0,0 +1,252 @@
+                                                                createdTopic ->
+                                                                        createdTopic
+                                                                                .name()
+                                                                                .equals(
+                                                                                        domainTopic
+                                                                                                .name()))
+                                                        .findFirst();
+
+                                        // is there an action to carry out
+                                        var createdStatus =
+                                                shouldCreate.isPresent()
+                                                        ? CRUD.CREATE
+                                                        : CRUD.IGNORED;
+                                        // if not ignored - was it created or updated
+                                        createdStatus =
+                                                createdStatus != CRUD.IGNORED
+                                                                && shouldCreate.isPresent()
+                                                                && wasCreated.isPresent()
+                                                        ? CRUD.CREATED
+                                                        : CRUD.CREATE;
+
+                                        status.crud(createdStatus)
+                                                .configs(domainTopic.configs())
+                                                .partitions(domainTopic.numPartitions())
+                                                .replication(domainTopic.replicationFactor());
+                                        if (exception != null) {
+                                            status.crud(CRUD.FAILED);
+                                            status.exception(exception);
+                                        }
+                                        return status.build();
+                                    })
+                            .collect(Collectors.toMap(TopicStatus::name, item -> item));
+        }
+    }
+    /** Topic provisioning status */
+    @Builder
+    @Data
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    @NoArgsConstructor(access = AccessLevel.PRIVATE)
+    @Accessors(fluent = true)
+    @SuppressFBWarnings
+    public static class TopicStatus {
+        private String name;
+        private CRUD crud;
+        private int partitions;
+        private short replication;
+        private Map<String, String> configs;
+        private Exception exception;
+    }
+
+    /** Operation result */
+    public enum CRUD {
+        CREATE,
+        CREATED,
+        READ,
+        UPDATE,
+        DELETE,
+        IGNORED,
+        FAILED
+    }
```

Comment:
lol - clearly not `CRUD` as its more than C,R,U,D.

Java docs required to explain the difference between `CREATE` and `CREATED` for sure!

Reply:
yeah - added the jdocs and renamed it to status